macros/parserMultiAnswer.pl

```diff
@@ -411,22 +411,21 @@ sub setMessage {
 ######################################################################
 
 #
-#  Produce the name for a named answer blank
-#  (Use the standard name for the first one, and
-#   create the prefixed names for the rest.)
+#  Produce the name for a named answer blank.
+#  (When the singleResult option is true, use the standard name for the first
+#  one, and create the prefixed names for the rest.)
 #
 sub ANS_NAME {
-  my $self = shift; my $i = shift; my $name;
+  my $self = shift; my $i = shift;
+  $self->{answerNames} = {} if !defined($self->{answerNames});
```

Comment:
This is not strictly necessary, since Perl will create the hash when it is first used.

Reply:
I wasn't sure on that.  I guess I should have tested that first.